src/internal/hooks/container-queries/use-resize-observer.ts

```diff
@@ -25,6 +25,12 @@ type ElementReference = (() => Element | null) | React.RefObject<Element>;
 export function useResizeObserver(elementRef: ElementReference, onObserve: (entry: ContainerQueryEntry) => void) {
   const stableOnObserve = useStableEventHandler(onObserve);
 
+  // HACK: Get the element during the render stage so that effects can be updated when this
+  //   changes. Ideally, we'd want to react to DOM changes and not tie element updates to React
+  //   state, but this fix makes the current approach a little bit safer.
+  const previousElement =
+    typeof window === 'undefined' ? null : typeof elementRef === 'function' ? elementRef() : elementRef?.current;
```

Comment:
Should the change of the referenced element be indicated by providing the new elementRef?